### PR TITLE
DF-2325- CF-Expandable Prototype

### PR DESCRIPTION
**Pull request by sebworks**
DF-2325- CF-Expandable Prototype - This a a basic View which could be substituted for an Ampersand-View, Backbone-View, or custom View. 

## Changes
- Updated `gulp/tasks/scripts.js` to precompile handlebars templates.
- Updated `package.json` to add necessary packages for prototype.
- Updated `src/static/css/cf-enhancements.less` to initial styles for expandable.
- Updated `src/static/css/nav-secondary.less` to fix issue with secondary nav on desktop.
- Updated `src/static/js/modules/beta-banner-state.js` to use Expandable view.
- Updated `src/static/js/modules/classes/BreakpointHandler.js` to pass back width argument into callback.
- Added `src/static/js/modules/classes/View.js` to create Backbone/Ampersand inspired View.
- Added `src/static/js/modules/expandable/ExpandableView.js` to create a Jquery free/test friendly Expandable prototype.
- Added `src/static/js/modules/expandable/expandableTemplate.hbs` as an example.
- Updated `src/static/js/modules/polyfill/.eslintrc` to add Function exception.
- Updated `src/static/js/modules/util/assign.js` to better reflect purpose.
- Updated `src/static/js/modules/secondary-nav-toggle.js` to refactor code to use Breakpoints argument and new expandable view.
- Added `src/static/js/modules/polyfill/function-bind.js` to add Function.bind polyfill.
- Updated `src/static/js/modules/util/date-range-formatter.js` to use assign method.
- Added `src/static/js/modules/util/extend.js` to extend objects and establish prototype chain.
- Added `src/static/js/modules/util/mixins/Events.js` to add basic event callback functionality.
- Added  `src/static/js/modules/util/result.js` as a general purpose property/function utility.
- Updated `src/static/js/routes/common.js` to remove `cf-expandable` and init `ExpandableView`

## Testing
-  Visit `http://localhost:7000/events/archive/` and click expandable. Observe interaction. 

## Notes 
- Will create a separate PR for unit tests.

## Review
@anselmbradford 
@jimmynotjim

### CODE REVIEW

File: src/static/js/modules/util/assign.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /* ==========================================================================
-   Shallow Extends
+   Assign
```
Comment (KimberlyMunoz) [inline src/static/js/modules/util/assign.js]: Should we include the original name and link to the repo so folks can find the original source code?

**Comment by jimmynotjim** [comment]
Sorry to throw a wrench in this late, but should this work have been done in cf-expandables itself (including unit tests)?

**Comment by sebworks** [comment]
That sounds good @jimmynotjim . I'll close these PRs and combine them.

File: src/static/js/modules/util/assign.js (hunk 1 above)
Comment (sebworks) [inline src/static/js/modules/util/assign.js]: Yep, wil make that change.

**Comment by sebworks** [comment]
https://github.com/cfpb/cfgov-refresh/pull/905#issuecomment-132827812

**Pull request closed without merging**